e2e/cypress/integration/login/tutorial_navigation_and_links_spec.js

```diff
@@ -0,0 +1,149 @@
+        checkStepOne();
+    }); // should go to first step
+
+    it('Clicking next on first step should show the second page', () => {
+        cy.get('#tutorialNextButton').click();
+        checkStepTwo();
+    }); // should go to second step
+
+    it('Clicking the icon should display the Mattermost download page.', () => {
+        // Source: https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/testing-dom__tab-handling-links/cypress/integration/tab_handling_anchor_links_spec.js
+        //
+        // This fetches the contents of the appDownloadImage link and checks it without actually
+        // clicking it and opening it in a new tab as Cypress doesn't appear to handle this.
+
+        cy.get('#appDownloadImage').then(($a) => {
+            const href = $a.prop('href');
+
+            // request the contents of https://www.google.com/
+            cy.request(href).
+
+            // drill into the response body
+                its('body').
+                should('include', 'https://mattermost.com/wp-content/themes/mattermostv3/img/android-icon.png').
+                should('include', 'https://mattermost.com/wp-content/themes/mattermostv3/img/google-play.png').
+                should('include', 'https://mattermost.com/wp-content/themes/mattermostv3/img/ios.png').
+                should('include', 'https://mattermost.com/wp-content/themes/mattermostv3/img/app-store.png').
+                should('include', 'https://mattermost.com/wp-content/themes/mattermostv3/img/windows-icon.png');
+        });
+    }); // click link for new tab
+
+    it('Clicking the Off-Topic link in sidebar should display that channel highlighted in the sidebar.', () => {
+        cy.get('#sidebarItem_off-topic').parent().should('not.have.class', 'active');
+        cy.get('#sidebarItem_off-topic').click();
+        cy.get('#sidebarItem_off-topic').parent().should('have.class', 'active');
+
+        //  Retest tutorial to ensure it has not changed:
+        checkStepTwo();
+    }); // click Off Topic link in sidebar
+
+    it('Clicking Next on page 2 should lead to page 3.', () => {
+        cy.get('#tutorialNextButton').click();
+        checkStepThree();
+    }); // click next to see step 3
+
+    it('Clicking Next on step 3 should hide the tutorial and show the previously selected Off Topic channel', () => {
+        cy.get('#tutorialNextButton').click();
+        checkOffTopicChannel();
+    }); // click to see Off Topic channel
+
+    it('Another user who views the tutorial and clicks Skip button should go right to the default channel.', () => {
+        cy.apiLogout();
+        cy.loginAsNewUserWithTutorial();
+        checkStepOne();
+        cy.get('#tutorialSkipLink').click();
+        checkTownSquare();
+    });
+}); // describe
+
+/*  This function checks the contents of the first step of the tutorial.
+*/
+
+function checkStepOne() {
+    cy.get('#tutorialIntroContent').should('be.visible').should('contain', 'Welcome to:');
```

Comment:
Since you are verifying the same element for all of these you should chain the additional assertions like 

```
cy.get('#tutorialIntroContent').
    should('be.visible').
    and('contain', 'Welcome to:').
    and('contain', 'Mattermost')
    // etc
```

Reply:
This applies to similar code below as well.

Reply:
Changed subsequent shoulds to ands.